src/main/java/com/artipie/settings/RepoData.java

```diff
@@ -0,0 +1,130 @@
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+public final class RepoData {
+    /**
+     * Repository settings create/read/update/delete.
+     */
+    private final CrudRepoSettings crs;
+
+    /**
+     * Ctor.
+     * @param crs Repository settings create/read/update/delete
+     */
+    public RepoData(final CrudRepoSettings crs) {
+        this.crs = crs;
+    }
+
+    /**
+     * Remove data from the repository.
+     * @param rname Repository name
+     * @return Completable action of the remove operation
+     */
+    public CompletionStage<Void> remove(final RepositoryName rname) {
+        final String repo = rname.toString();
+        return this.asto(rname).deleteAll(new Key.From(repo)).thenAccept(
+            nothing -> Logger.info(this, String.format("Removed data from repository %s", repo))
+        );
+    }
+
+    /**
+     * Move data when repository is renamed: from location by the old name to location with
+     * new name.
+     * @param rname Repository name
+     * @param nname New repository name
+     * @return Completable action of the remove operation
+     */
+    public CompletionStage<Void> move(final RepositoryName rname, final RepositoryName nname) {
+        final Key repo = new Key.From(rname.toString());
+        final Key nrepo = new Key.From(nname.toString());
+        final Storage asto = this.asto(rname);
+        return new SubStorage(repo, asto).list(Key.ROOT).thenCompose(
+            list -> new Copy(new SubStorage(repo, asto), list)
+                .copy(new SubStorage(nrepo, asto))
+        ).thenCompose(nothing -> asto.deleteAll(new Key.From(repo))).thenAccept(
+            nothing ->
+                Logger.info(this, String.format("Moved data from repository %s to %s", repo, nrepo))
+        );
+    }
+
+    /**
+     * Obtain storage from repository settings.
+     * @param rname Repository name
+     * @return Abstract storage
+     * @throws UncheckedIOException On IO errors
+     */
+    private Storage asto(final RepositoryName rname) {
+        try {
+            final YamlMapping yaml = this.crs.valueAsYaml(rname);
+            YamlMapping res = yaml.yamlMapping("storage");
+            if (res == null) {
+                res = this.storageYamlByAlias(rname, yaml.string("storage"));
+            }
+            return new YamlStorage(res).storage();
+        } catch (final IOException err) {
+            throw new UncheckedIOException(err);
+        }
+    }
+
+    /**
+     * Find storage settings by alias, considering two file extensions and two locations.
+     * @param rname Repository name
+     * @param alias Storage settings yaml by alias
+     * @return Yaml storage settings found by provided alias
+     * @throws IllegalStateException If storage with given alias not found
+     * @throws UncheckedIOException On IO errors
+     * @checkstyle LineLengthCheck (2 lines)
+     */
+    private YamlMapping storageYamlByAlias(final RepositoryName rname, final String alias) {
```

Comment:
call `com.artipie.settings.StorageAliases#find` to obtain `StorageAliases` instance, then create `StorageYamlConfig` instance using `StorageAliases` and yaml settings and call `StorageYamlConfig#storage` to get a storage